instrumentation/kafka-clients/src/main/java/brave/kafka/clients/KafkaHeaders.java

```diff
@@ -17,12 +17,17 @@
 import org.apache.kafka.common.header.Header;
 import org.apache.kafka.common.header.Headers;
 
+import static brave.kafka.clients.KafkaTracing.log;
 import static java.nio.charset.StandardCharsets.UTF_8;
 
 final class KafkaHeaders {
   static void replaceHeader(Headers headers, String key, String value) {
-    headers.remove(key);
-    headers.add(key, value.getBytes(UTF_8));
+    try {
+      headers.remove(key);
+      headers.add(key, value.getBytes(UTF_8));
+    } catch (IllegalStateException e) {
```

Comment:
let's catch throwable vs guess which exceptions might be raised during this? less maintenance that way.

```suggestion
    } catch (Throwable t) {
      propagateIfFatal(t);
```

Reply:
You did see that `headers.remove` and `headers.add` are both declared as `throws IllegalStateException`? Though since that's not a checked exception I can see the argument that it's not actually easy to maintain.